### ISSUE TITLE
maptrV2 Prepare nuScenes data centerline

### ISSUE THREAD

**Issue by ldhai** [hustvl/MapTR#93]
When I use the code will generate nuscenes_map_infos_temporal_{train,val}.pkl, which contain local vectorized map annotations. I meet an error ("networkx.exception.NodeNotFound: target node [(15.0, 2.355), (15.0, -2.636), (4.167, -30.0), (-15.0, 15.813), (0.284, -30.0)] not in graph"). Has anyone encountered this error and how to solve it?
![image](https://github.com/hustvl/MapTR/assets/9911239/846e939d-86fa-4f58-b8ca-42561d9453ef)

**Comment by LegendBC** [comment]
Other people have contacted me about this error. But I rerun my script in my environment and didn't reproduce this error. Can you provide me with your specific environmental setting?

I uploaded the annotation file in the repo's GitHub releasement. You can directly try MapTRv2 with our generated annotation files.

**Comment by yqj20020174** [comment]
I have the same problem. I cannot find the annotation file. Where is it ?

**Comment by ldhai** [comment]
> I have the same problem. I cannot find the annotation file. Where is it ?

https://github.com/hustvl/MapTR/releases

**Comment by ldhai** [comment]
> Other people have contacted me about this error. But I rerun my script in my environment and didn't reproduce this error. Can you provide me with your specific environmental setting?
> 
> I uploaded the annotation file in the repo's GitHub releasement. You can directly try MapTRv2 with our generated annotation files.

I'm running up training with your annotation file, can you provide me with your trained MapTRv2* (with centerline) model, I'd like to visualize the effect with your model first. My environment is as follows, that error above, I don't know if it's a mistake in the code or step where the data is processed:

![image](https://github.com/hustvl/MapTR/assets/9911239/d6e04855-c8e7-4d7e-ad92-3621fd725e10)
![image](https://github.com/hustvl/MapTR/assets/9911239/0086857e-1c9e-41a0-8eb2-e2665eaa2374)
![image](https://github.com/hustvl/MapTR/assets/9911239/e10fa4cd-d645-422e-83bb-805a16da23a5)
![image](https://github.com/hustvl/MapTR/assets/9911239/3910e18a-6f6d-47e1-857e-62c55e2a9db2)

**Comment by LegendBC** [comment]
@ldhai I just upload centerline ckpt & log, please check it out.

**Comment by ldhai** [comment]
> @ldhai I just upload centerline ckpt & log, please check it out.

Thank you very much! Hmmmmm, can you update that visualization code, it's the maptrV2 visualization code (with the centerline), or is it enough to just use the maptr visualization code.

**Comment by chdzhen** [comment]
when I used v1.0-mini. have same error, can you help me to generate the nuscenes_map_infos_temporal_{train,val}.pkl of v1.0-mini.

**Comment by chdzhen** [comment]
> 

at file 'custom_nusc_map_converter.py'   line 742, need add "for leave in leaves:"  ?
 for root in roots:
    for leave in leaves:
        paths = nx.all_simple_paths(pts_G, root, leave)
        all_paths.extend(paths)

**Comment by chdzhen** [comment]
> Other people have contacted me about this error. But I rerun my script in my environment and didn't reproduce this error. Can you provide me with your specific environmental setting?
> 
> I uploaded the annotation file in the repo's GitHub releasement. You can directly try MapTRv2 with our generated annotation files.

at file 'custom_nusc_map_converter.py' line 742, need add "for leave in leaves:" ?
for root in roots:
for leave in leaves:
paths = nx.all_simple_paths(pts_G, root, leave)
all_paths.extend(paths)

**Comment by pnnnnnnn** [comment]
i have the same problem. i update networkx from 2.2 to 2.3 and fix this problem

**Comment by LegendBC** [comment]
> > @ldhai I just upload centerline ckpt & log, please check it out.
> 
> Thank you very much! Hmmmmm, can you update that visualization code, it's the maptrV2 visualization code (with the centerline), or is it enough to just use the maptr visualization code.

Yes, the visualization is almost the same. You only need to add one more color for centerline and use `plt.quiver`.

**Comment by LegendBC** [comment]
> > Other people have contacted me about this error. But I rerun my script in my environment and didn't reproduce this error. Can you provide me with your specific environmental setting?
> > I uploaded the annotation file in the repo's GitHub releasement. You can directly try MapTRv2 with our generated annotation files.
> 
> at file 'custom_nusc_map_converter.py' line 742, need add "for leave in leaves:" ? for root in roots: for leave in leaves: paths = nx.all_simple_paths(pts_G, root, leave) all_paths.extend(paths)

You don't need nested for loop and you can refer to official documents for further details.

**Comment by LegendBC** [comment]
> i have the same problem. i update networkx from 2.2 to 2.3 and fix this problem

Thanks for your help and remind. My networkx version is 2.8.7

**Comment by MonkeyJohn** [comment]
> i have the same problem. i update networkx from 2.2 to 2.3 and fix this problem

well , it works

**Comment by HongbiaoZ** [comment]
> @ldhai I just upload centerline ckpt & log, please check it out.

Hi, thanks for your great work. I didn't find the link of ckpt and log file for maptrv2 in Readme. Could you send the link here?

**Comment by PeggyPeppa** [comment]
@LegendBC Hi Can you provide the whole enviroment of yor our device? It seems that there are alot conflicts according to your installation readme.

**Comment by Ricky-Developer** [comment]
> i have the same problem. i update networkx from 2.2 to 2.3 and fix this problem

thanks a lot!

**Comment by liaoqingPKU** [comment]
> > @ldhai I just upload centerline ckpt & log, please check it out.
> 
> Hi, thanks for your great work. I didn't find the link of ckpt and log file for maptrv2 in Readme. Could you send the link here?



> > @ldhai I just upload centerline ckpt & log, please check it out.
> 
> Hi, thanks for your great work. I didn't find the link of ckpt and log file for maptrv2 in Readme. Could you send the link here?



> > @ldhai I just upload centerline ckpt & log, please check it out.
> 
> Hi, thanks for your great work. I didn't find the link of ckpt and log file for maptrv2 in Readme. Could you send the link here?

change to main branch and you will find the links
